### PR TITLE
Fix linter warnings

### PR DESCRIPTION
**Pull request by bastelfreak**
This is a test for automatically fixing pep8 warnings

### CODE REVIEW

**Comment by bastelfreak** [comment]
Hi,

could somebody review this please? Any Idea why there are still D203 warnings even if ignore it in tox.ini?

CC @heg-hpo @pbyatshon

File: marmoset/dhcp/dhcp_config.py [hunk 1]
```diff
@@ -17,8 +17,14 @@ def __init__(self, mac, ip_address, gateway=None, networkmask=None):
 
         self.set_settings(True, mac, ip_address, gateway, networkmask)
 
-    def set_settings(self, allow_none_value_for_not_required_parameter=False, mac=None, ip_address=None,
-                     gateway=None, networkmask=None):
+    def set_settings(
+            self,
+            allow_none_value_for_not_required_parameter=False,
```
Comment (taxus13) [inline marmoset/dhcp/dhcp_config.py:22]: is ist possible to use a shorter variable here? maybe "allow_optional_none_values" or sth like that?

File: marmoset/cli/installimage_parser.py [hunk 2]
```diff
@@ -5,22 +5,27 @@ def add_to(parser, name, **kwargs):
...
-    installimage_list.set_defaults(func=installimage.list)
-    installimage_remove = subcommands.add_parser('remove',
-                                                 help='remove a installimage config for a MAC address',
-                                                 aliases=['r', 'delete', 'del', 'd'])
+    installimage_list = subcommands.add_parser(
+        'list',
+        help='list MAC addresses for all currently present Installimage configs',
+        aliases=['l'])
+    installimage_list.set_defaults(func=installimage.dolist)
+    installimage_remove = subcommands.add_parser(
```
Comment (taxus13) [inline marmoset/cli/installimage_parser.py:26]: in line 8ff you have one argument per line, this should be consitent here

File: marmoset/cli/pxe_parser.py [hunk 3]
```diff
@@ -23,19 +26,21 @@ def add_to(parser, name, **kwargs):
...
+    pxe_list = subcommands.add_parser(
+        'list',
+        help='list IP addresses for all currently present PXE client config',
+        aliases=['l'])
+    pxe_list.set_defaults(func=pxe.dolist)
 
-    pxe_remove = subcommands.add_parser('remove',
-                                        help='remove a PXE config for an IP address',
-                                        aliases=['r', 'delete', 'del', 'd'])
+    pxe_remove = subcommands.add_parser(
```
Comment (taxus13) [inline marmoset/cli/pxe_parser.py:40]: in line 34ff you have one argument per line, this should be consitent here

File: marmoset/dhcp/isc_dhcp_ldap_config.py [hunk 4]
```diff
@@ -29,49 +29,67 @@ def __get_server_connection():
     def save(self):
         conn = self.__get_server_connection()
 
-        dhcpStatements = ["fixed-address %s;" % self.dhcp_config.ip_address,
```
Comment (taxus13) [inline marmoset/dhcp/isc_dhcp_ldap_config.py:32]: personnally i find the old version here more readable than the new one...

File: marmoset/installimage/installimage_config.py [hunk 5]
```diff
@@ -57,9 +61,9 @@ def __read_config_file(self, path=None):
 
         lines = []
 
-        with open(path, 'r') as f:
-            lines = f.readlines()
-            f.close()
+        with open(path, 'r') as file:
+            lines = file.readlines()
+            file.close()
```
Comment (taxus13) [inline marmoset/installimage/installimage_config.py]: when you use the "with open" function, you do not have to close the file explicit:
https://docs.python.org/3.5/tutorial/inputoutput.html#methods-of-file-objects (last example of that section)

File: marmoset/installimage/installimage_config.py [hunk 6]
```diff
@@ -86,7 +91,9 @@ def __write_config_file(self, path=None):
...
 
+        # pylint is wrong on this one as it seems
+        #pylint: disable-msg=unexpected-keyword-arg
         os.makedirs(InstallimageConfig.CFG_DIR, exist_ok=True)
-        with open(path, 'w') as f:
-            f.write(content)
-            f.close()
+        with open(path, 'w') as file:
+            file.write(content)
+            file.close()
```
Comment (taxus13) [inline marmoset/installimage/installimage_config.py]: same as in line 66

File: marmoset/pxe/client_config.py [hunk 7]
```diff
@@ -156,14 +159,15 @@ def __write_config_file(self, content, path=None):
...
 
+        # pylint seems to be wrong on this one
+        #pylint: disable-msg=unexpected-keyword-arg
         os.makedirs(ClientConfig.CFG_DIR, exist_ok=True)
-        f = open(path, 'w')
-        f.write(content)
-        f.close()
+        file = open(path, 'w')
+        file.write(content)
+        file.close()
```
Comment (taxus13) [inline marmoset/pxe/client_config.py]: use the "with open" function here, is it good practice according to the python documentation:
https://docs.python.org/3.5/tutorial/inputoutput.html#methods-of-file-objects (last example)

**Comment by taxus13** [comment]
I added some comments, maybe they are helpful

File: marmoset/installimage/installimage_config.py (hunk 5 above)
Comment (bastelfreak) [inline marmoset/installimage/installimage_config.py]: fixed, commit is f663d07

File: marmoset/installimage/installimage_config.py (hunk 6 above)
Comment (bastelfreak) [inline marmoset/installimage/installimage_config.py]: fixed in a8ca375

File: marmoset/dhcp/dhcp_config.py (hunk 1 above)
Comment (bastelfreak) [inline marmoset/dhcp/dhcp_config.py:22]: The name is a bit long, yes. But IMO it describes very good the usage of the variable. Not sure if we want to rename that. Any other opinions here?

File: marmoset/cli/installimage_parser.py (hunk 2 above)
Comment (bastelfreak) [inline marmoset/cli/installimage_parser.py:26]: fixed in 7ad9e0a

File: marmoset/cli/pxe_parser.py (hunk 3 above)
Comment (bastelfreak) [inline marmoset/cli/pxe_parser.py:40]: fixed in e781da1

File: marmoset/dhcp/isc_dhcp_ldap_config.py (hunk 4 above)
Comment (bastelfreak) [inline marmoset/dhcp/isc_dhcp_ldap_config.py:32]: as far as I can tell the line was too long for the linter.

File: marmoset/pxe/client_config.py (hunk 7 above)
Comment (bastelfreak) [inline marmoset/pxe/client_config.py]: implemented in b578413

**Comment by bastelfreak** [comment]
We're down to 16 warnings, I would like to disable the D203 stuff in tox but that didn't work out. Any ideas? Also any ideas about the other errors?

File: marmoset/dhcp/dhcp_config.py (hunk 1 above)
Comment (sebastianrakel) [inline marmoset/dhcp/dhcp_config.py:22]: what's wrong with long variables names? it's full describes the usage, so no possibility of misunderstanding

File: marmoset/dhcp/isc_dhcp_ldap_config.py (hunk 4 above)
Comment (sebastianrakel) [inline marmoset/dhcp/isc_dhcp_ldap_config.py:32]: i commited something of both worlds :D PEP8 says max line length is 80 chars (IMHO 80 chars as guideline, should be deprecated)

with the commit c5f0698cb377c9824f2d1cb7bde6282b24fa88ab
it's under 80 chars but more readable

**Pull request merged**